themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -36,18 +36,19 @@
             alt="{$product.name}"
           {/if}
           loading="lazy"
-          width="452"
-          height="452"
+          width="{$product.default_image.bySize.medium_default.width}"
```

Comment:
MMM I wonder why do you use the `medium_default` sizes when the image in the source is `large_default` ?
Shouldn't you use the sizes from the type of image that is used?

Reply:
@jolelievre Matthieu went for this solution because he thinks the image quality drops. It can happen and is very real, especially when you are using mac with a retina display.

However, I think the right way would be to use medium_default image, with correct dimensions. If we want to improve quality on scaled displays, we can increase JPG compression or use 2x source-sets.

Reply:
Yep I agree, but maybe overkill for a simple fix 🤔. Let's keep this simple for now But having multiple source could be a nice new feature to integrate